zip-0316.rst

```diff
@@ -599,26 +649,157 @@ to give access only to view incoming payments (as opposed to change).
+Typecode; a Consumer MUST reject as invalid any UA/UVK that violates
+this requirement.
 
 Since Metadata Items are not Receivers, they MUST NOT be selected by
 a Sender when choosing a Receiver to send to, and since they are not
 Viewing Keys, they MUST NOT provide additional authority to view
 information about transactions.
 
-Currently no Metadata Types are defined. New Metadata Types SHOULD be
-introduced either by a modification to this ZIP or by a new ZIP, in
-accordance with the ZIP Process [#zip-0000]_.
+New Metadata Types SHOULD be introduced either by a modification to this
+ZIP or by a new ZIP, in accordance with the ZIP Process [#zip-0000]_.
 
+Rationale for making Revision 0 UA/UVKs with MUST-understand Typecodes invalid
+''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+.. raw:: html
+
+   <details>
+   <summary>Click to show/hide</summary>
+
+A Consumer implementing this ZIP prior to `Revision 1`_ will not
+recognize the Human-Readable Parts beginning with “``ur``” that mark a
+`Revision 1`_ UA/UVK. So if a UA/UVK that includes MUST-understand
+Typecodes is required to use these `Revision 1`_ HRPs, this will ensure
+that the MUST-understand specification is correctly enforced even for
+such implementations.
+
+.. raw:: html
+
+   </details>
+
+
+Address Expiration Metadata
+---------------------------
+
+As of `Revision 1`_, Typecodes :math:`\mathtt{0xE0}` and :math:`\mathtt{0xE1}`
+are reserved for optional address expiry metadata. A producer MAY choose to
+generate Unified Addresses containing either or both of the following Metadata
+Item Types, or none.
+
+The value of a :math:`\mathtt{0xE0}` item MUST be an unsigned 32-bit integer in
+little-endian order specifying the Address Expiry Height, a block height of the
+Zcash chain associated with the UA/UVK. A Unified Address containing metadata
+Typecode :math:`\mathtt{0xE0}` MUST be considered expired when the height of
+the Zcash chain is greater than this value.
+
+The value of a :math:`\mathtt{0xE1}` item MUST be an unsigned 64-bit integer in
+little-endian order specifying a Unix Epoch Time, hereafter referred to as the
+Address Expiry Time. A Unified Address containing Metadata Typecode
+:math:`\mathtt{0xE1}` MUST be considered expired when the current time is
+after the Address Expiry Time.
+
+A Sender that supports `Revision 1`_ of this specification MUST set
+a non-zero ``nExpiryHeight`` field in transactions it creates that are
+sent to a Unified Address that defines an Address Expiry Height. If the
+``nExpiryHeight`` normally constructed by the Sender would be greater than the
+Address Expiry Height, then the transaction MUST NOT be sent. If only an
+Address Expiry Time is specified, then the Sender SHOULD choose a value for
+``nExpiryHeight`` such that the transaction will expire no more than 24 hours
+after the current time. If both :math:`\mathtt{0xE0}` and :math:`\mathtt{0xE1}`
```

Comment:
> after the current time

If the address expires 30 *days* from now, why should a transaction expire within 24 hours of when it is sent? Shouldn't we say the transaction should expire no later than 24 hours before the address expiration time?